hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoTransferHandler.java

```diff
@@ -616,10 +629,100 @@ public Fees calculateFees(@NonNull final FeeContext feeContext) {
         return feeContext
                 .feeCalculator(subType)
                 .addBytesPerTransaction(bpt)
+                .addRamByteSeconds(associationsRbs)
                 .addRamByteSeconds(rbs * USAGE_PROPERTIES.legacyReceiptStorageSecs())
                 .calculate();
     }
 
+    private long getAutoAssociationsRbsFee(final FeeContext feeContext) {
+        final var body = feeContext.body();
+        final var op = body.cryptoTransferOrThrow();
+        final var tokenTransfers = op.tokenTransfers();
+
+        // If the token transfers collection is empty, then we have only hbars transfer and no token
+        // associations are needed, so we do not need to calculate the next part.
+        if (tokenTransfers.isEmpty()) {
+            return 0;
+        }
+
+        final var accountStore = feeContext.readableStore(ReadableAccountStore.class);
+        final var tokenRelStore = feeContext.readableStore(ReadableTokenRelationStore.class);
+
+        final Set<AccountID> recipientsSet = getTransferRecipients(tokenTransfers);
+        final HashMap<AccountID, HashSet<TokenID>> associationsMap = getTokenTransferAssociations(tokenTransfers);
+
+        var associationsRbs = 0;
+        for (var accountId : recipientsSet) {
+            var account = accountStore.getAccountById(accountId);
+            // If this is a transfer to an alias and the account is null this is a hollow account creation case,
+            // and we need to set default values.
+            if (account == null && accountId.hasAlias()) {
+                var numAutoAssociations = associationsMap.get(accountId).size();
```

Comment:
One suggestion here for better readability here might be:
```
if(isHollow(account)) {
     accociationsRbs += calculateHollowAccountRbs(params)
} else {
     accociationsRbs += calculateAccountRbs(params)
} 
```

and then in those 2 new methods extracts the calculation logics.
This could improve code readability.

Reply:
Comment addressed partially. I did not use `isHollow` because if I pass just the account it can be null and I can't check for its alias. If I pass just the accountId then I need to fetch the account from the store twice and it seems redundant. To create an `isHollow` method I need to pass both the account and the accountId and this would seem strange to me in terms of readability.